worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelDistributor.java

```diff
@@ -122,49 +129,45 @@ public ShovelDistributor(
     }
     
     public void run() throws InterruptedException {
-        try {
-            while(true) {
-                runOnce();
-
-                try {
-                    Thread.sleep(distributorRunIntervalMilliseconds);
-                } catch (final InterruptedException e) {
-                    LOGGER.warn("Interrupted {}", e.getMessage());
-                    throw e;
-                }
-            }
-        } finally {
-            try {
-                nonRunningShovelCheckerExecutorService.shutdownNow();
-            } catch (final Exception exception) {
-                LOGGER.warn("Failed to shutdown nonRunningShovelCheckerExecutorService", exception);
-            }
+        while(true) {
+            runOnce();
 
             try {
-                nonRunningShovelCheckerExecutorService.shutdownNow();
-            } catch (final Exception exception) {
-                LOGGER.warn("Failed to shutdown shovelRunningTooLongCheckerExecutorService", exception);
-            }
-
-            try {
-                corruptedShovelCheckerExecutorService.shutdownNow();
-            } catch (final Exception exception) {
-                LOGGER.warn("Failed to shutdown corruptedShovelCheckerExecutorService", exception);
+                Thread.sleep(distributorRunIntervalMilliseconds);
+            } catch (final InterruptedException e) {
+                LOGGER.warn("Interrupted {}", e.getMessage());
+                throw e;
             }
         }
     }
     
     public void runOnce() {
-        
-        final Set<DistributorWorkItem> distributorWorkItems = getDistributorWorkItems();
+
+        final Set<DistributorWorkItem> distributorWorkItems;
+        try {
+            distributorWorkItems = getDistributorWorkItems();
+        } catch (final Exception e) {
+            final String errorMessage = String.format(
+                    "Failed to get a list of distributor work items, so unable to check if any shovels need to be created. " +
+                            "Will try again during the next run in %d milliseconds",
+                    distributorRunIntervalMilliseconds);
+
+            LOGGER.error(errorMessage, e);
+
+            return;
```

Comment:
If we get an error when querying the RabbitMQ API for shovels, or when creating a shovel, we catch the exception, log an error, and return.

For consistency, I updated the call to get a list of queues (`getDistributorWorkItems`) to have the same behavior. Previously, the exception would propagate and crash the application. There is no benefit to crashing, so I am catching and logging here instead.